client/src/components/ui/button.tsx

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+import * as React from "react";
+import { Slot } from "@radix-ui/react-slot";
+import { cva, type VariantProps } from "class-variance-authority";
+
+import { cn } from "~/lib/utils";
+import Loader from "../loader";
+
+const buttonVariants = cva(
+  cn(
+    "inline-flex items-center justify-center whitespace-nowrap rounded-lg text-sm font-medium ring-offset-background transition-colors w-fit",
+    "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2", // focus
+    "disabled:pointer-events-none disabled:opacity-70", // disabled
+  ),
+  {
+    variants: {
+      variant: {
+        default: "bg-primary text-primary-foreground hover:bg-primary/80",
+        outline: "border border-border/15 hover:bg-border/10 ",
+        secondary: "bg-secondary text-secondary-foreground hover:bg-card",
+        success:
+          "bg-success text-success-foreground hover:bg-success/75 outline outline-1 outline-success/30 border-[3px] border-background",
+        successOutline:
+          "border border-success text-success hover:bg-success/30",
+        warning:
+          "bg-warning text-warning-foreground hover:bg-warning/75 outline outline-1 outline-warning/30 border-[3px] border-background",
+        warningOutilne:
+          "border border-warning text-warning hover:bg-warning/30",
+        destructive:
+          "bg-destructive text-destructive-foreground hover:bg-destructive/80 outline outline-1 outline-destructive/30 border-[3px] border-background",
+        destructiveOutilne:
+          "border border-destructive text-destructive hover:bg-destructive/30",
+        ghost: "hover:bg-secondary/30",
+        link: "text-primary underline-offset-4 hover:underline",
+      },
+      size: {
+        default: "h-10 px-4 py-2",
+        sm: "h-9 px-3",
+        lg: "h-11 px-8",
+        icon: "h-10 w-10 rounded-full",
+      },
+    },
+    defaultVariants: {
+      variant: "default",
+      size: "default",
+    },
+  },
+);
+
+const commonStyle =
```

Comment:
`commonStyle` is only used once. from its name I understand that it should be reused in the button variants? if not it doenst have to exist, you can inline the classname on line 112

Reply:
removed